src/main/java/sa/assertj/maps/ContainsAnyOf.java

```diff
@@ -0,0 +1,47 @@
+package sa.assertj.maps;
+
+import sa.assertj.Experiment;
+import sa.assertj.Util;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ContainsAnyOf extends Experiment {
+
+    static DataProvider provider = (size, numOfSamples) -> {
+        Object[][] result = new Object[numOfSamples][];
+        for (int s=0; s < numOfSamples; s++) {
+            Map<String, String> actual = new LinkedHashMap<>();
+            Map.Entry[] expected = new Map.Entry[size];
+            for (int i=0; i < size; i++) {
+                String key = Util.randomString(10);
+
+                if (actual.containsKey(key)) {
+                    i--;
+                    continue;
+                }
+
+                String value = Util.randomString(10);
+
+                actual.put(key, value);
+
+                // Only make last one be present in map
+                if (i < size - 1) {
```

Comment:
If the map you are using is unsorted than this is fine but if it is sorted, we should choose a random index at the start which we will actually put in.